lib/net.js

```diff
@@ -1174,11 +1176,27 @@ function Server(options, connectionListener) {
     configurable: true, enumerable: false
   });
 
+  Object.defineProperty(this, '_handle', {
+    set: function(val) {
+      if (val === null) {
+        this[serverListenHasBeenCalled] = false;
+      }
+      this[serverHandle] = val;
+    },
+    get: function() {
+      return this[serverHandle];
+    },
+    enumerable: true
+  });
```

Comment:
The main reason I decided to do it in this way it's because we need to set `serverListenHasBeenCalled` back to false when we close connection or some error has happened. And I have noticed that for both cases we call the same code to handle `_handle` field. Concretely:

`    this._handle.close();
    this._handle = null;` 

So in order to avoid repetitions and do something like this every time we close this._handle

 `   
    this._handle.close();
    this._handle = null;
    this.serverListenHasBeenCalled = false
` 

I decided to make this stuff in one place.

A big plus of this, is that if someone in a future will add some logic to handle error, they will not forget to set `this.serverListenHasBeenCalled = false`.